.github/workflows/gcs-benchmark.yml

```diff
@@ -0,0 +1,49 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# A workflow to trigger gcs tests
+name: GCS Benchmark Test
+
+on:
+  workflow_dispatch:
+
+jobs:
+  Benchmark:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: run benchmark
+        env:
+          PROJECT: rapids-spark
+          DATAPROC_REGION: us-central1
+          COMPUTE_REGION: us-central1
+          COMPUTE_ZONE: us-central1-a
+          GCS_BUCKET: spark-rapids-ml-benchmarking
+          KEY_FILE_CONTENT: ${{ secrets.GCLOUD_PRIVATE_KEY }}
+          SERVICE_ACCOUNT: ${{ secrets.GCLOUD_SERVICE_ACCOUNT }}
+        shell: bash
+        run: |
+          set -x
+          cat <<< $KEY_FILE_CONTENT > key.json
+          gcloud auth activate-service-account $SERVICE_ACCOUNT --key-file key.json
+          gcloud config set project $PROJECT
+          gcloud config set dataproc/region $DATAPROC_REGION
+          gcloud config set compute/region $COMPUTE_REGION
+          gcloud config set compute/zone $COMPUTE_ZONE
+          export BENCHMARK_HOME=$GCS_BUCKET/benchmark
+          cd python/benchmark/dataproc
+          ./setup.sh
+          ./run_benchmark.sh
```

Comment:
- Wondering if we should gather/report the timings somewhere?  i.e. Spark-RAPIDS has a regular perf/regression report.  (Could be deferred to future work).
- Should we explicitly shutdown the Dataproc cluster (vs. waiting for 30m idle timeout)?  Not sure if we'd want to leave it up for a bit just for debugging/troubleshooting purposes or not.

Reply:
Nice PR. Also interested if we should gather timings or assert the speedup is larger than a threshold.

Reply:
For the first point, maybe we can just capture the logs per the README (assuming the output of the `egrep` shows up in the build logs):
```bash
./run_benchmark.sh gpu 2>&1 | tee benchmark.log
egrep -e "[0-9.]* seconds" *.out
```

For the second point, was mostly referring to this part of the README:
```bash
gcloud dataproc clusters delete ${USER}-spark-rapids-ml-[cpu|gpu] --region ${COMPUTE_REGION}
```

Otherwise, the cluster will eventually [time out after 30m of idle time](https://github.com/NVIDIA/spark-rapids-ml/blob/branch-23.08/python/benchmark/dataproc/start_cluster.sh#L64).

Reply:
What's next:
**log**
capture the logs into a file and store on Jenkins, so that users can download from Jenkins build.

**timeout**
Since you've defined cluster timeout in shell script, I only need to define a timeout operation (delete the cluster and stop the pipeline) in Jenkins script in case the pipeline gets stuck.

**notification**
Email or Slack channel?

**schedule**
weekly? nightly?

Reply:
Sounds good.  As for the run schedule and notification preferences, it sounds like we'd be OK with a weekly run with a slack notification.

Reply:
>What's next:
log
capture the logs into a file and store on Jenkins, so that users can download from Jenkins build.

We upload jenkins log to github for external contributor to access, why we do this? The log is accessible for all users if its already in github....

Reply:
> timeout

gcp should have provide the way to delete the cluster automatically, just try https://cloud.google.com/dataproc/docs/reference/rest/v1/ClusterConfig#lifecycleconfig

![Uploading image.png…]()

Reply:
I think default email is entirely fine (just thought slack was the convention).  Also, the current cluster config should delete itself after 30m of idle per link I posted above.

Reply:
sure, we can let it auto-delete when idle for 30 mins, if we want to save more money then we could try decrease it later~

Reply:
I've added the schedule option, it will auto-trigger every Monday 1PM UTC. Seems we don't need more changes of log, timeout and notification.